### ISSUE TITLE
Possible wrong error return value

### ISSUE THREAD

**Issue by db-tech** [ultravideo/uvgRTP#36]
While working a little bit on #33 I've found some strange error return values on in different places:

```
        return -RTP_RECV_ERROR;
        return -RTP_RECV_ERROR;
        return -RTP_INVALID_VALUE;
        return -RTP_INVALID_VALUE;
        return -RTP_NOT_SUPPORTED;
        return -RTP_INVALID_VALUE;
```

Could this just be wrong / refactoring error or do i missunderstand something ? Then you can just close this.

**Comment by altonen** [comment]
Hi,

these have been taken out of context so I can't say with 100% certainty but I believe they are correct. Some internal functions of uvgRTP return negative error codes as positive return values are used to signal something else (# of bytes IIRC). Please reopen this issue if you find them to be actually invalid.

**Comment by db-tech** [comment]
Just wondered why they have a minus in  front? You can search for those, I believe it's Just in two files.

**Comment by polusto** [comment]
I noticed the same. They might work correctly, but they are very confusing. Why not just return the actual enum value the minus corresponds to?

**Comment by altonen** [comment]
I rechecked and they are indeed incorrect. I think what happened was that originally uvgRTP's error codes were positive values (so `RTP_GENERIC_ERROR` was 1 instead of -1) and when I  needed to return # of bytes/error code, I had to negate the `rtp_error_t` to indicate to the caller that an error had occurred. Then at some point I needed positive status codes that signaled non-error status other than `RTP_OK` such as `RTP_PTK_READY` and had to make the actual error codes negative. I forgot to update these return values and remove the minus sign.
